fili-core/src/test/groovy/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandlerSpec.groovy

```diff
@@ -0,0 +1,545 @@
+        writer.writeValueAsString(weightQuery) >> { throw new JsonProcessingException("word") }
+        CacheWeightCheckRequestHandler handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        ) {
+            @Override
+            public SuccessCallback buildCacheSuccessCallback(
+                    final SuccessCallback classicSuccessCallback,
+                    final CacheService cacheService,
+                    final DruidAggregationQuery<?> groupByQuery,
+                    final ResponseProcessor response
+            ) {
+                groupByQuery.clone()
+                return success
+            }
+        }
+
+        1 * queryWeightUtil.skipWeightCheckQuery(groupByQuery) >> false
+        0 * next.handleRequest(context, request, groupByQuery, response) >> true
+        1 * queryWeightUtil.getQueryWeightThreshold(DAY) >> 5
+        1 * groupByQuery.clone() >> groupByQuery
+        1 * webService.postDruidQuery(context, success, null, null, weightQuery)
+        1 * response.getErrorCallback(groupByQuery)
+        1 * response.getFailureCallback(groupByQuery)
+
+        expect:
+        handler.handleRequest(context, request, groupByQuery, response)
+
+        and:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 1
+    }
+
+    def "Test handle request on cache hit responds to the group by request"() {
+        setup:
+        GroupByQuery groupByQuery = RequestUtils.buildGroupByQuery()
+        QuerySigningService querySigningService = Mock(SegmentIntervalsHashIdGenerator)
+        querySigningService.getSegmentSetId(_) >> Optional.of(1234L)
+        ContainerRequestContext containerRequestContext = Mock(ContainerRequestContext)
+        containerRequestContext.getHeaders() >> (["Bard-Testing": "###BYPASS###", "ClientId": "UI"] as
+                MultivaluedHashMap<String, String>)
+        RequestContext requestContext = new RequestContext(containerRequestContext, true)
+        requestContext.isReadCache() >> true
+        MAPPER.valueToTree(groupByQuery) >> json
+        cacheService = new CacheService(dataCache, querySigningService, MAPPER)
+        CacheWeightCheckRequestHandler handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        )
+
+        expect: "The count of fact query cache hit is 0"
+        bardQueryInfo.queryCounter.get(BardQueryInfo.FACT_QUERY_CACHE_HIT).get() == 0
+
+        when: "A groupBy query runs with a valid cache hit"
+        boolean requestProcessed = handler.handleRequest(requestContext, request, groupByQuery, response)
+
+        then: "Check the cache and return valid json"
+        1 * dataCache.get(_) >> new MemTupleDataCache.DataEntry<String>("key1", 1234L, "[]")
```

Comment:
this is good test code for the cache service, probably can be simplified out of the requestHandler testing